_posts/2024-01-16-owasp-top10-with-abstract-examples-part-2.md

```diff
@@ -0,0 +1,120 @@
+
+## 10 Server-Side Request Forgery
+
+[Server-side request forgery](https://owasp.org/Top10/A10_2021-Server-Side_Request_Forgery_%28SSRF%29/) (SSRF) occurs when the server fails to validate the user-supplied URL to an external resource. A bad request can go to some third-party service or local file system with extended permissions.
+
+Let's assume that the customer could influence the kitchen to get a steak from the more expensive restaurant next door. The steak is then paid for cheaper and the restaurant loses money. It would also be bad if a customer in the driving lane directs the payment request to the next customer in line and speeds away.
+
+### The fix
+
+The server must sanitize all input, perhaps use whitelists for secure connections, should not send raw responses to clients (may contain something useful to attacker), and use firewall rules to prevent unintended connections. This should give us better control over what the user is doing.
+
+In our example scenario, the restaurant's kitchen staff must confirm the request made to the kitchen. If a steak was requested from the restaurant next door, the request should set off alarm bells in the heads of the kitchen staff. Regarding the redirection of the payment request. The cashier does not have to accept the current customer's comment "the next customer in line pays".
+
+## Conclusion
+
+Thank you for reading this blog post. The topic is important and will never go away. Although technical fixes themselves are a practical way to raise the level of information security, the best fix right now is to improve the information security culture of organizations. This ultimately leads to improved security at all levels.
+
+If we understand the most common risks of web applications and projects overall, we have a better chance of fixing them in time. Fixing them in time also means saving money. Once the risks mentioned above are minimized, we can focus more [on other growing problems](https://yle.fi/a/74-20065352) such as phishing.
+
+Phishing is a huge problem, and it keeps getting worse because of new large language model tools. Did you know that [there has been 1265% increase of malicious phishing emails](https://www.cnbc.com/2023/11/28/ai-like-chatgpt-is-creating-huge-increase-in-malicious-phishing-email.html) since fourth quarter of 2022? Phishing is still a good attack vector, because it is so effective. 
+
+Don't cut corners. Neglecting data security will hit harder later. Someone may think that the damage caused by a breach can be avoided only by paying the fines received for it. However, it is worth remembering that a data breach can affect more than just the company’s business. For example, in [Vastaamo's data breach](https://en.wikipedia.org/wiki/Vastaamo_data_breach), customers' really sensitive data became public knowledge, and people lost trust in information systems, perhaps forever.
```

Comment:
voisi myös mainita, että GDPR voi antaa sakkoja, jos et suojele asiakkaiden dataa oikein https://gdpr-info.eu/issues/fines-penalties/

Reply:
Ykkösosan johdannossa yritin pelotella jonkuverran, että myös non-tech ihmiset kiinnostuis tästä blogikirjotuksesta :) Kirjoitin sinne mm näin:

"These days, companies have to deal with regulatory fines, legal costs, operational downtime, and last but not least, reputational damage. In the worst case scenario, the organization's information can be encrypted, inaccessible to the organization, or eventually completely public."

Voisin tuon sun linkin lisätä tuohon kappaleeseen.